osu.Game.Rulesets.Catch/UI/CatchPlayfield.cs

```diff
@@ -29,6 +29,8 @@ public class CatchPlayfield : ScrollingPlayfield
 
         internal readonly CatcherArea CatcherArea;
 
+        private CatchComboDisplay comboDisplay => CatcherArea.ComboDisplay;
```

Comment:
this should not be dragged here out of the catcher area like this - `GameplayBeatmap`'s `LastJudgementResult` was designed for this sort of use case (albeit right now it seems to lack support for reverting, but I'm not sure that's too much of an issue, can even be done as part of this PR most likely).

Reply:
I did it like this because the judgement events are in here, and because I didn't want to really pass the judgements through each component's `OnNewResult` and `OnRevertResult` (yeah `CatcherArea` would have to make a revert result method).

Didn't know about `LastJudgementResult`, TIL, will use it.

Reply:
checked myself - using `LastJudgementResult` in its current state is infeasible because (a) no support for reverting and (b) information about the DHO is not in `JudgementResult`, which is needed in this case for combo colour, so for now I just moved the calls to the counter's `On{New,Revert}Result` to `CatcherArea`. reasoning is that it's pretty WTF for `CatcherArea`'s `OnNewResult` method update only *half* of its children.

this could be improved upon if `GameplayBeatmap` had a way to proxy `On{New,Revert}Result` entirely as an event instead of just exposing the last judgement result. unsure if that's what we want, however.